docs/logging-configuration.asciidoc

```diff
@@ -0,0 +1,105 @@
+ifdef::env-github[]
+NOTE: For the best reading experience,
+please view this documentation at https://www.elastic.co/guide/en/apm/agent/ruby[elastic.co]
+endif::[]
+
+[[logging-configuration]]
+== Logging configuration
+
+The Ruby agent allows you to set logging options as well as configure trace/log correlation.
+Please see the <<configuration,configuration documentation>> to learn how to set <<config-logger,a logger>>, <<config-log-path,
+a log path>> and <<config-log-level,log level>>.
+
+[float]
+[[trace-log-correlation]]
+==== Trace/log correlation
+
+Trace/log correlation can be set up in three different ways.
+
+[float]
+[[rails-tagged-logging]]
+==== Rails TaggedLogging
+
+Rails applications configured with an `ActiveSupport::TaggedLogging` logger can append the correlation IDs to log output.
+Add the following to the Rails environment configuration file:
+
+[source,ruby]
+----
+config.logger = ActiveSupport::TaggedLogging.new(my_logger)
+config.log_tags = [proc { ElasticAPM.log_ids }]
+
+# Logs will then include the correlation IDs:
+#
+# [transaction.id=c1ae84c8642891eb trace.id=b899fc7915e801b7558e336e4952bafe] Started GET "/" for 127.0.0.1 at 2019-09-16 11:28:46 +0200
+# [transaction.id=c1ae84c8642891eb trace.id=b899fc7915e801b7558e336e4952bafe] Processing by ApplicationController#index as HTML
+# [transaction.id=c1ae84c8642891eb trace.id=b899fc7915e801b7558e336e4952bafe]   Rendering text template
+# [transaction.id=c1ae84c8642891eb trace.id=b899fc7915e801b7558e336e4952bafe]   Rendered text template (Duration: 0.1ms | Allocations: 17)
+# [transaction.id=c1ae84c8642891eb trace.id=b899fc7915e801b7558e336e4952bafe] Completed 200 OK in 1ms (Views: 0.4ms | Allocations: 171)
+----
+**Note:** Because of the order in which Rails computes the tags for logs and executes the request, the span id might not be included.
+Consider using `Lograge` instead, as the timing of its hooks allow the span id to be captured in logs.
+
+[float]
+[[lograge]]
+==== Lograge
+
+With `lograge` enabled and set up in your Rails application, modify the `custom_options` block in the Rails environment
+configuration file. The returned `Hash` will be included in the structured Lograge logs.
+
+[source,ruby]
+----
+config.lograge.custom_options = lambda do |event|
+  ElasticAPM.log_ids do |transaction_id, span_id, trace_id|
+    { transaction_id: transaction_id,
```

Comment:
Remember to change the keys to dot-scoped `transaction.id`

Reply:
oh, good catch. Thanks!

Reply:
https://github.com/elastic/apm-agent-ruby/pull/527/commits/173f0b0787e2f949aa206ca17ef5b83852cb3f10